application/clicommands/ProcessCommand.php

```diff
@@ -192,9 +193,10 @@ protected function renderProblemTree($tree, $useColors = false, $depth = 0, BpNo
             $node = $subtree['node'];
             $state = $parent !== null ? $parent->getChildState($node) : $node->getState();
 
+            $colors = [];
             if ($node instanceof HostNode) {
                 $colors = $this->hostColors[$state];
-            } else {
+            } elseif ($node instanceof ServiceNode) {
                 $colors = $this->serviceColors[$state];
             }
 
```

Comment:
I think after this if-elseif, there is an if block which apply colors if the `--colors` flag is used in the process command. I think there, you would also have to check if the `$colors` is not an empty array.